schoolmatch.py

```diff
@@ -13,152 +20,171 @@
 import pandas as pd
 import numpy as np
 
-OSPI_df = pd.read_csv('Washington_School_Directory_20171121.csv')
-OSPI_Ind_df = pd.read_csv('ApprovedPrivateSchoolsList.csv')
 
-RS_df_pre = pd.read_csv('RSSchoolList2.csv', usecols=[0,1,2,3,4])
-RS_df_NotPub = RS_df_pre[RS_df_pre.Type != "Public"]
-RS_df = RS_df_pre[RS_df_pre.Type == "Public"]
+def clean_title(text):
+    """
+    Takes a pandas series of strings, and returns a pandas series of strings
+    that have been stripped and each word capitalized
+    """
+    text = text.str.lower()
+    text = text.str.title()
+    text = text.str.strip()
+    return text
+
+
+def string_truncate(strings_to_truncate, words_to_remove, truncated_list_name):
+    """
+    Takes a list of strings to modify, a list of words to remove from the first list,
+    and an empty list that will be populated with with the new truncated strings.
+    """
+    truncated_list_name = []
+    for string in strings_to_truncate:
+        modifying_string = string
+        for word in words_to_remove:
+            modifying_string = modifying_string.replace(word, "")
+            modifying_string = modifying_string.strip()
+        truncated_list_name.append(modifying_string)  
+    return truncated_list_name
+#Will take a look at stop words next
+
+
+#read in all csvs and create relevant dataframes
+ospi_raw_pub_df = pd.read_csv('Washington_School_Directory_20171121.csv')
+ospi_raw_ind_df = pd.read_csv('ApprovedPrivateSchoolsList.csv')

```

Comment:
I hadn't every had encoding problems with any of my work coding or otherwise before this. This was the one problem I ran into that I just wanted to disappear (clearly I'm not alone given article titles like "CSV: An Encoding Nightmare" and "Escaping from character encoding hell..." ha. I thought I had saved the appropriate files in the repo, but apparently not. In this first run, I think I was hoping to do all the cleaning programmatically. I've given up an inch there and just fixed it in excel.